semgrep/tests/performance/test_semgrep_rules_repo_perf.py

```diff
@@ -221,20 +221,27 @@
 
 
 @pytest.fixture(scope="session")
-def semgrep_rules_repo(tmp_path_factory):
+def semgrep_rules_repo(request, tmp_path_factory):
```

Comment:
So, I'm kinda confused about the architecture here. From what I understand about this PR, we're performing these fixture operations when tests are *collected*, not when they're *run*. This gives us a performance hit because these fixtures take a while to run. Is that right?

Based on above, I have a few questions:

* Are these fixtures being run because they don't have the `benchmark` argument/fixture? That's why adding it below causes the fixtures to be avoided unless we're actually benchmarking?
* Why do we have two separate fixtures here that only appear to be used with each other? Could this be one fixture to simplify things?
* It seems we could leverage existing pytest functionality rather than checking this information ourselves. Whether that's something like adding `benchmark` here to only run this fixture during benchmarking, or leveraging `pytest.mark.parametrize` with `indirect=True` to run the fixture during test-setup instead of test-collection. Am I missing something that makes that not possible?

Reply:
Basically I think all the questions are answered by: I assumed skipped tests' fixtures would not be executed. Sorry about that :(

This PR is the fastest way I found to fix the CI failures that are repo-wide, cause I'm supposed to dedicate all my time to semgrep-app now. The alternative I see is reverting the benchmarks commit.